.github/workflows/tag-release.yml

```diff
@@ -4,33 +4,74 @@ name: Create release tag
+    paths:
+      - emscripten-releases-tags.json
+      - .github/workflows/tag-release.yml
+    branches:
+      - main
   workflow_dispatch:
 
 jobs:
-    tag-release:
-      # Only activate for commits created by the create-release.yml workflow.
-      # The assumption is that when manual changes happen, we want to handle
-      # tagging manually too.
-      if: github.event.head_commit.author.username == 'github-actions[bot]'
-      runs-on: ubuntu-latest
-      steps:
-        - name: Match message and create tag
-          uses: actions/github-script@v7
-          with:
-            script: |
-              const message = `${{ github.event.head_commit.message }}`
-              const regex = /Release ([0-9]+.[0-9]+.[0-9]+)/
-              const match = message.match(regex)
-              if (match) {
-                const release = match[1]
-                console.log(`Matched release ${release}`)
-                await github.rest.git.createRef({
-                  owner: context.repo.owner,
-                  repo: context.repo.repo,
-                  ref: `refs/tags/${release}`,
-                  sha: context.sha
-                })
-              } else {
-                console.log(`Commit message: ${message} did not match pattern`)
-              }
+  tag-release:
+    # Only activate for commits created by the create-release.yml workflow.
+    # The assumption is that when manual changes happen, we want to handle
+    # tagging manually too.
+    name: Check for release commit and create tag
+    if: github.event.head_commit.author.username == 'github-actions[bot]'
+    runs-on: ubuntu-latest
+    outputs:
+      is_release: ${{ steps.create-tag.outputs.result }}
+    steps:
+      - name: Match message and create tag
+        id: create-tag
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.TEST_TOKEN }}
+          # A commit with the message of the form 'Release X.Y.Z' is expected
+          # to have been created by create_release.py and update the latest
+          # release in emscripten-releases-tags.json
+          script: |
+            const message = `${{ github.event.head_commit.message }}`
+            const regex = /Release ([0-9]+.[0-9]+.[0-9]+)/;
+            const match = message.match(regex);
+            let is_release = false;
+            if (match) {
+              const release = match[1];
+              console.log(`Matched release ${release}`);
+              await github.rest.git.createRef({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                ref: `refs/tags/${release}`,
+                sha: context.sha
+              });
+              is_release = true;
+            } else {
+              console.log(`Commit message: '${message}' did not match pattern`);
+            }
+            return is_release;
+
+  dispatch-emscripten-tag:
```

Comment:
What triggers this job?

Reply:
This job has a 'needs' clause on line 56 which means it depends on the tag-release job (so if that job fails or doesn't run, then this one won't run). If it didn't have that then IIUC all the jobs in the workflow would run in parallel.